src/read.ts

```diff
@@ -0,0 +1,121 @@
+import Mute from 'mute-stream'
+import { Completer, AsyncCompleter, createInterface, ReadLineOptions } from 'readline'
+
+export namespace Read {
+  export interface Options<T extends string | number = string> {
+    default?: T
+    input?: ReadLineOptions['input'] & {
+      isTTY?: boolean
+    }
+    output?: ReadLineOptions['output'] & {
+      isTTY?: boolean
+    }
+    prompt?: string
+    silent?: boolean
+    timeout?: number
+    edit?: boolean
+    terminal?: boolean
+    replace?: string,
+    completer?: Completer | AsyncCompleter,
+  }
+}
+
+export default async function read<T extends string | number = string> ({
```

Comment:
As discussed in [the tshy readme](https://github.com/isaacs/tshy#handling-default-exports), default exports are a royal pain for hybrid modules.

My recommendation is to make this a named export, and just accept `import { read } from 'read'` as the way to import it.

If it really truly _must_ be a default export, then some hacking is required.

Add `".": "./src/index.ts"` to the `tshy.exports` config.

Create a `src/index.ts` that does `export * from './read.js'`. This is the ESM main export.

Create `src/index-cjs.cts` containing:

```ts
import * as items from './read.js'
declare global {
   namespace Read {
    export type Options = items.Options
  }
}
export = items.read
```

Then anyone using cjs will get the types from the `Read.Options` global namespace (and pray that no one else tries to put something there!) and they will get `read()` as the actual `module.exports` returned by require (instead of require() returning `{ default: read }`).

But again, really, it's much better to be able to `import { type Options, read } from 'read'` and have it work the same way for both without any extra work.